File: code/Iniciante/p-level-1/1021.java
/*
Nome:      Notas e Moedas
ID:        1021
Resposta:  Accepted
Linguagem: Java
Tempo:     0.068s
Tamanho:   1,93 KB
Submissao: 06/06/16 14:17:47
*/
import java.io.IOException;
import java.util.Scanner;
import java.util.Locale;
 
/**
 * IMPORTANT: 
 *      O nome da classe deve ser "Main" para que a sua solução execute
 *      Class name must be "Main" for your solution to execute
 *      El nombre de la clase debe ser "Main" para que su solución ejecutar
 */
public class Main {
 
	private static class Troco {
		/**
		 *	Recebe como entrada um valor de ponto flutuante
		 *  representando uma quantidade de dinheiro. Então
		 *  imprime o número mínimo de notas e moedas para 
		 *  representar essa quantia.
		 */
		private double dinheiro;
		private int[] valorNota, valorMoeda;
		
		public Troco(double dinheiro) {
			this.dinheiro = dinheiro;
			this.valorNota = new int[6];
			this.valorNota[0] = 10000;
			this.valorNota[1] = 5000;
			this.valorNota[2] = 2000;
			this.valorNota[3] = 1000;
			this.valorNota[4] = 500;
			this.valorNota[5] = 200;
			this.valorMoeda = new int[6];			
			this.valorMoeda[0] = 100;
			this.valorMoeda[1] = 50;
			this.valorMoeda[2] = 25;
			this.valorMoeda[3] = 10;
			this.valorMoeda[4] = 5;
			this.valorMoeda[5] = 1;
		}
		
		public void troco() {
			int dinheiro = (int)(this.dinheiro * 100);
			System.out.println("NOTAS:");
			for(int i = 0; i < 6; i++) {
				int qtdNotas = dinheiro/valorNota[i];
				dinheiro %= valorNota[i];				
				System.out.format(Locale.US, "%d nota(s) de R$ %.2f\n", qtdNotas, valorNota[i]/100.0);
			}
			System.out.println("MOEDAS:");
			for(int i = 0; i < 6; i++) {
				int qtdMoeda = dinheiro/valorMoeda[i];
				dinheiro %= valorMoeda[i];
				System.out.format(Locale.US, "%d moeda(s) de R$ %.2f\n", qtdMoeda, valorMoeda[i]/100.0);
			}
		}
	}
 
    public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		double N = Double.parseDouble(sc.next());
		Troco mostraTroco = new Troco(N);
		mostraTroco.troco();
		sc.close();
    }
 
}

File: code/Iniciante/p-level-1/1021.cpp
/*
Nome:      Notas e Moedas
ID:        1021
Resposta:  Accepted
Linguagem: C++
Tempo:     0.000s
Tamanho:   1,08 KB
Submissao: 04/01/15 20:24:49
*/
#include <stdio.h>
int main(){
double N;
int N100, N50, N20, N10, N5, N2, C100, C50, C25, C10, C5, C1, NN;
scanf("%lf",&N);
NN = N * 100;
N100 = NN / 10000;
NN = NN % 10000;
N50 = NN / 5000;
NN = NN % 5000;
N20 = NN / 2000;
NN = NN % 2000;
N10 = NN / 1000;
NN = NN % 1000;
N5 = NN / 500;
NN = NN % 500;
N2 = NN / 200;
NN = NN % 200;
    printf("NOTAS:\n");
    printf("%d nota(s) de R$ 100.00\n",N100);
    printf("%d nota(s) de R$ 50.00\n",N50);
    printf("%d nota(s) de R$ 20.00\n",N20);
    printf("%d nota(s) de R$ 10.00\n",N10);
    printf("%d nota(s) de R$ 5.00\n",N5);
    printf("%d nota(s) de R$ 2.00\n",N2);
C100 = NN / 100;
NN = NN % 100;
C50 = NN / 50;
NN = NN % 50;
C25 = NN / 25;
NN = NN % 25;
C10 = NN / 10;
NN = NN % 10;
C5 = NN / 5;
NN = NN % 5;
C1 = NN;
    printf("MOEDAS:\n");
    printf("%d moeda(s) de R$ 1.00\n",C100);
    printf("%d moeda(s) de R$ 0.50\n",C50);
    printf("%d moeda(s) de R$ 0.25\n",C25);
    printf("%d moeda(s) de R$ 0.10\n",C10);
    printf("%d moeda(s) de R$ 0.05\n",C5);
    printf("%d moeda(s) de R$ 0.01\n",C1);
}
